src/samplebuffer.cpp

```diff
@@ -0,0 +1,21 @@
+#include "samplebuffer.h"
```

Comment:
I don't expect any severe performance impacts by not inlining the constructor/destructor. So I preferred to hide the implementation details. The implementation of alloc/free will be moved to SampleBuffer.cpp anyway after all unmanaged arrays have been eliminated. At least then the constructor and destructor will no longer be inlined.

I added the clear() and fill() operations just for convenience. Their invocation is currently not time critical, they are currently used only for initialization.